fmralign/_utils.py

```diff
@@ -168,10 +210,21 @@ def _make_parcellation(imgs, clustering, n_pieces, masker, smoothing_fwhm=5, ver
             else:
                 parcellation = Parcellations(
                     method=clustering, n_parcels=n_pieces, mask=masker, verbose=verbose)
-        parcellation.fit(imgs)
-        labels_img = parcellation.labels_img_
+        parcellation.fit(images_to_parcel)
+        labels = _apply_mask_fmri(
+            parcellation.labels_img_, masker.mask_img_).astype(int)
+
     else:
         raise InputError(
             ('Clustering should be "kmeans", "ward", "rena", "hierarchical_kmeans", \
-             or a 3D Niimg'))
-    return _apply_mask_fmri(labels_img, masker.mask_img_).astype(int)
+             or a 3D Niimg and n_pieces an integer ≥ 1'))
```

Comment:
Light suggestion for clarity:

```suggestion
             or a 3D Niimg, and n_pieces should be an integer ≥ 1'))
```